### ISSUE TITLE
import/no-named-as-default: rule never fires (named exports of the imported module are not resolved), causing live eslint-disable directives to be reported as unused

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25624]
## Summary

The native `import/no-named-as-default` rule never reports a violation when a default import reuses the name of the imported module's named export, because oxlint does not resolve the imported module's named-export set. This silently disables the rule, and under `--report-unused-disable-directives` it mis-flags genuinely-needed `eslint-disable` directives as "unused" — actively pushing maintainers to delete directives that are still live. The rule name is accepted by the config parser (a bogus name under the `import` plugin is rejected), so the rule exists yet emits nothing.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`PortalPaywall.tsx` (exports the SAME identifier both as a named export and as the default):

```tsx
export const PortalPaywall = () => null
export default PortalPaywall
```

`PortalPaywall.stories.tsx` (default-imports it using that same identifier — the exact `no-named-as-default` pattern):

```tsx
// eslint-disable-next-line import/no-named-as-default
import PortalPaywall from './PortalPaywall'
```

`.oxlintrc.json`:

```json
{
	"plugins": ["import"],
	"categories": {"correctness": "off"},
	"rules": {"import/no-named-as-default": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --report-unused-disable-directives PortalPaywall.stories.tsx
```

(Also reproduces without the directive, on a plain-JS variant, and when the sibling `PortalPaywall.tsx` is included in the same run.)

## Expected

An `import/no-named-as-default` diagnostic on the `import PortalPaywall from './PortalPaywall'` line: `PortalPaywall` is also a _named_ export of `./PortalPaywall`, so using it as the default-import binding is the precise construct this rule forbids. ESLint 7 and ESLint 9 (eslint-plugin-import@2.31.0 with a working TS resolver) both flag it at 1:8 (`Using exported name 'PortalPaywall' as identifier for default import`); consequently the `eslint-disable-next-line` directive is LIVE and must NOT be reported as unused. The reference rule resolves the imported module and checks its export map (`ExportMap.get(source, context)` → `importedModule.has(localDefaultName)`); oxlint's implementation appears not to resolve the sibling module's named exports.

## Actual

oxlint 1.78.0 reports no `import/no-named-as-default` diagnostic at all (exit 0). With `--report-unused-disable-directives` it instead emits:

```
PortalPaywall.stories.tsx:1:1: warning: Unused eslint-disable directive (no problems were reported).
```

## Cross-checked

| Linter | Result on the import (no directive) | Directive fixture + `--report-unused-disable-directives` |
| --- | --- | --- |
| ESLint 7.23.0 + eslint-plugin-import@2.31.0 (+ @typescript-eslint/parser@6.21.0) | flagged at 1:8, exit 1 | clean, exit 0 — directive live |
| ESLint 9.39.4 (flat config) + eslint-plugin-import@2.31.0 (+ typescript-eslint@8.62.1 parser) | flagged at 1:8, exit 1 | clean, exit 0 — directive live |
| oxlint 1.78.0 | no diagnostic, exit 0 | directive reported unused at 1:1 |
